### PR TITLE
Simplify BufferedSeriesIterator usage

### PR DESCRIPTION
**Pull request by free**
Allow for `BufferedSeriesIterator` instances to be created without an underlying iterator, to simplify their usage.

Signed-off-by: Alin Sinpalean <alin.sinpalean@gmail.com>

### CODE REVIEW

**Comment by brian-brazil** [comment]
That looks cleaner (and will save a bit of memory for federation). We can merge this after the bugfix moratorium, and you've fixed the failing tests.

**Comment by free** [comment]
Sorry for the tests, I had copied a test file into the root dir and was getting a confusing build failure for it (which I've now figured out) and had ascribed the real test failures to the same mystery cause. Fixed now.

**Comment by brian-brazil** [comment]
Don't worry about it, it happens to us all.

File: storage/buffer.go [hunk 1]
```diff
@@ -26,9 +26,25 @@ type BufferedSeriesIterator struct {
...
 }
 
+type nopIterator struct{}
+
+func (nopIterator) At() (int64, float64) { return math.MinInt64, 0 }
+func (nopIterator) Seek(t int64) bool    { return false }
+func (nopIterator) Next() bool           { return false }
+func (nopIterator) Err() error           { return nil }
+
+var nopIt = nopIterator{}
```
Comment (tomwilkie) [inline storage/buffer.go]: You could put this in `storage/noop.go` with the other no-op implementation of the storage interfaces...

Comment (free) [reply storage/buffer.go]: Done.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**